### PR TITLE
Feature/tvist1 394 notify boardmembers

### PR DESCRIPTION
**Pull request by jekuaitk**
* Allows broadcasting agendas to assigned board members.

https://jira.itkdev.dk/browse/TVIST1-394

### CODE REVIEW

File: src/Controller/AgendaBroadcastController.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            foreach ($boardMembers as $boardMember) {
+                $boardMemberAddress = $cprHelper->getAddressFromCpr($boardMember->getCpr());
+
+                $digitalPostRecipients[] = (new DigitalPost\Recipient())
+                    ->setName($boardMember->getName())
+                    ->setIdentifierType('CPR')
+                    ->setIdentifier($boardMember->getCpr())
+                    ->setAddress($boardMemberAddress)
+                ;
+            }
```
Comment (rimi-itk) [inline src/Controller/AgendaBroadcastController.php:95]: We probably need to handle exceptions when getting the board member address.

File: templates/agenda/broadcast/create.html.twig [hunk 2]
```diff
@@ -8,18 +8,16 @@
     <div class="row">
         <div class="col">
             <h1>{% trans %}Broadcast agenda{% endtrans %}</h1>
-            <p class="text-muted"><small>{% trans %}Send notice to board members informing them that they now has access to the agenda.{% endtrans %}</small></p>
+            <p class="text-muted"><small>{% trans %}Send notice to board members informing them that they now have access to the agenda.{% endtrans %}</small></p>
```
Comment (rimi-itk) [inline templates/agenda/broadcast/create.html.twig:11]: 🎉

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. We need to handle errors when getting board member addresses.

And we need to decide if we need physical addresses on board members (cf. https://github.com/itk-dev/naevnssekretariatet/pull/196#issuecomment-1091668116)

File: src/Controller/AgendaBroadcastController.php (hunk 1 above)
Comment (jekuaitk) [reply src/Controller/AgendaBroadcastController.php]: Handled exceptions in 830e7bd1942f6fc8d52396e7bdc1bf1cbd9b654e.

File: src/Controller/AgendaBroadcastController.php [hunk 3]
```diff
@@ -58,6 +62,36 @@ public function broadcastAgenda(Agenda $agenda, CprHelper $cprHelper, DigitalPos
...
+                    ;
+                } catch (CprException $e) {
+                    switch (true) {
+                        case $e->getPrevious() instanceof NoPnrFoundException:
+                            $this->addFlash('error', $translator->trans('PNR for {boardMember} not found ({cprNumber})', ['boardMember' => $boardMember->getName(), 'cprNumber' => $boardMember->getCpr()], 'digital_post'));
+                            break;
+                        case $e->getPrevious() instanceof ServiceException:
+                            $this->addFlash('error', $e->getPrevious()->getMessage());
+                            break;
+                    }
```
Comment (rimi-itk) [inline src/Controller/AgendaBroadcastController.php:87]: This is the best use of a `switch` statement I've seen in a long time. 🎉

File: templates/agenda/broadcast/index.html.twig [hunk 4]
```diff
@@ -5,6 +5,11 @@
 {% block title %}{% trans %}Agenda broadcasts{% endtrans %}{% endblock %}
 
 {% block header %}
+    {% for message in app.flashes('error') %}
+        <div class="alert alert-danger">
+            {{ message }}
+        </div>
+    {% endfor %}
```
Comment (rimi-itk) [inline templates/agenda/broadcast/index.html.twig:12]: We should move this to our base template and handle all types of flash messages (cf. https://symfony.com/doc/current/controller.html#flash-messages).

**Review by rimi-itk — CHANGES_REQUESTED** [review]
If possible, the display of flash messages should be moved to a base template.

File: templates/agenda/broadcast/index.html.twig (hunk 4 above)
Comment (jekuaitk) [reply templates/agenda/broadcast/index.html.twig]: Moved to `templates/layout-with-agenda-submenu.html.twig`.

File: src/Controller/AgendaBroadcastController.php [hunk 5]
```diff
@@ -79,10 +79,10 @@ public function broadcastAgenda(Agenda $agenda, CprHelper $cprHelper, DigitalPos
                 } catch (CprException $e) {
                     switch (true) {
                         case $e->getPrevious() instanceof NoPnrFoundException:
-                            $this->addFlash('error', $translator->trans('PNR for {boardMember} not found ({cprNumber})', ['boardMember' => $boardMember->getName(), 'cprNumber' => $boardMember->getCpr()], 'digital_post'));
+                            $this->addFlash('danger', $translator->trans('PNR for {boardMember} not found ({cprNumber})', ['boardMember' => $boardMember->getName(), 'cprNumber' => $boardMember->getCpr()], 'digital_post'));
                             break;
                         case $e->getPrevious() instanceof ServiceException:
-                            $this->addFlash('error', $e->getPrevious()->getMessage());
+                            $this->addFlash('danger', $e->getPrevious()->getMessage());
                             break;
```
Comment (rimi-itk) [inline src/Controller/AgendaBroadcastController.php:86]: You could consider using a `match` expression (https://www.php.net/manual/en/control-structures.match.php) to get the message, e.g..

```php
$message = match (true) {
  $e->getPrevious() instanceof NoPnrFoundException => $translator->trans('PNR for {boardMember} not found ({cprNumber})', ['boardMember' => $boardMember->getName(), 'cprNumber' => $boardMember->getCpr()], 'digital_post'),
  $e->getPrevious() instanceof ServiceException => $e->getPrevious()->getMessage(),
  default => $e->getMessage(),
};
$this->addFlash('danger', $message);
```

**Review by rimi-itk — APPROVED** [review]
Approved with a suggestion for a possible improvement.

File: src/Controller/AgendaBroadcastController.php (hunk 5 above)
Comment (jekuaitk) [reply src/Controller/AgendaBroadcastController.php:86]: 👍

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**